runtime-builder/go-build.sh

```diff
@@ -29,18 +29,45 @@ fi
 export PATH=/usr/local/go/bin:"${PATH}"
 export GOPATH="${workspace}"/_gopath
 
-# Move application files into a temporary staging directory, dependencies excluded.
-staging=$(mktemp -d staging.XXXX)
-find . -mindepth 1 -maxdepth 1 ! -name "${staging}" ! -name "$(basename $GOPATH)" -exec mv {} "${staging}"/ \;
+# $workspace contains files and directories in original app.yaml directory. It
+# consists of the main package files as well as application resource files.
+# If there are dependency packages, $workspace will also contain _gopath
+# directory containing these with the proper GOPATH structure.
+
+
+# If there is _gopath/main-package-path file, reconstruct the main path relative
+# to GOPATH/src by moving files over.
+mainpath=""
+if [[ -f "${GOPATH}/main-package-path" ]]; then
+  mainpkg=$(cat ${GOPATH}/main-package-path)
+  if [[ "$mainpkg" != "" ]]; then
+    mainpath="${GOPATH}/src/${mainpkg}"
+    mkdir -p "${mainpath}"
+    for f in $(find . -mindepth 1 -maxdepth 1 ! -name "$(basename $GOPATH)")
```

Comment:
Should there be anything other than the file "mainpkg" itself?

Reply:
Not sure I understand.  Are you referring to $mainpath directory instead?  $mainpath directory may contain dependency packages (including internal and vendor), these will also reside in $workspace and hence I'm simply moving what's in $workspace into $mainpath.  While files can simply be mv over, if there are existing directories, mv will error, and hence I need to rm first.